### PR TITLE
Maintenance: update --with-gnutls detection

### PR DESCRIPTION
**Pull request by yadij**
Simplify the SQUID_CHECK_LIB_WORKS(gnutls,...) logic
to use all the latest abilities of that macro

### CODE REVIEW

**Comment by rousskov** [comment]
> Simplify the SQUID_CHECK_LIB_WORKS(gnutls,...) logic to use all the latest abilities of that macro added in PR #1684

It may be a good idea to [fix](https://github.com/squid-cache/squid/actions/runs/7951993897/job/21706074247?pr=1685#step:4:12125) this PR build before merging #1684 (and before posting even more PRs that depend on #1684), just in (unlikely) case the build failures in this PR are related to some (unknown) problems in #1684 rather than problems in this PR itself.

File: configure.ac [hunk 1]
```diff
@@ -1077,37 +1077,16 @@ dnl Solaris10 provides MD5 natively through libmd5
...
-    SSLLIB="$LIBGNUTLS_PATH $LIBGNUTLS_LIBS $SSLLIB"
-    AC_DEFINE(USE_GNUTLS,1,[GnuTLS support is available])
-  ])
+  PKG_CHECK_MODULES([LIBGNUTLS],[gnutls >= 3.4.0],[],[:])
+  CPPFLAGS="$LIBGNUTLS_PATH $CPPFLAGS"
+  AC_CHECK_HEADERS(gnutls/gnutls.h gnutls/x509.h gnutls/abstract.h)
+  SQUID_STATE_ROLLBACK(squid_gnutls_state)
 ])
 
+SSLLIB=""
```
Comment (rousskov) [inline configure.ac:1089]: AFAICT, prior to this PR, SSLLIB contained either GnuTLS or OpenSSL library. After this PR, SSLLIB becomes specific to OpenSSL. After this PR is merged, will you post an OpenSSL-specific PR that renames SSLLIB?

**Review by rousskov — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (yadij) [reply configure.ac:1089]: yes, that is the plan.

File: src/security/forward.h [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #include "base/ToCpp.h"
 #include "security/LockingPointer.h"
 
-#if USE_GNUTLS && HAVE_GNUTLS_ABSTRACT_H
+#if HAVE_GNUTLS_ABSTRACT_H
```
[the remark this answers is not in the document]
Comment (yadij) [reply src/security/forward.h:17]: I would rather not. Removing this has revealed some bugs in the `configure.ac` logic around header detection when the library is auto-disabled.

[the remark this answers is not in the document]
Comment (rousskov) [reply src/security/forward.h:17]: That is not a valid excuse to stop keeping GnutTLS-specific code inside USE_GNUTLS guard (or its replacement) IMO.

[the remark this answers is not in the document]
Comment (yadij) [reply src/security/forward.h:17]: reverted while resolving conflicts.

**Review by rousskov — COMMENTED** [review]
I cannot efficiently review most of this PR changes, unfortunately.

I (still) do not recommend using GitHub PR numbers in commit messages. We should be using commit SHAs instead IMO.

Current PR description also implies that we are simplifying SQUID_CHECK_LIB_WORKS logic "to use ...". The intent, AFAICT, was to say that we can now simplify that logic "by using ...":

> PR description: 

```diff
- to use all the latest abilities of that macro added in PR #1684
+ by using that macro improvements added in commit bf2e100.
```

**Review by kinkie — APPROVED** [review]

**Pull request closed without merging**